code/assignTime.m

```diff
@@ -162,25 +178,73 @@
     chunkIndices{1} = 1:numPackets;
 end
 
+%%
+% Two types of chunks -- those which come after gaps < 6 seconds (as determined by
+% timestamp) and those which come after gaps > 6 seconds (potential for complete
+% roll-over of systemTick). For chunks which follow a gap < 6 seconds,
+% use systemTick to continue calculation of DerivedTime after the gap (rather
+% than creating a new correctedAlignTime for that chunk); still honor 1/Fs
+% spacing of DerivedTime. Note -- if new chunk is created because of change
+% in sampling rate, calculate a new correctedAlignTime for that chunk
+
+% Only need to do this calculation if more than 1 chunk
+numChunks = length(chunkIndices);
+chunksWithTimingFromPrevious = [];
+if numChunks > 1
+    % Get timestamps of first and last packet in chunks to calculate time gaps
+    for iChunk = 1:numChunks
+        indices_FirstPacket(iChunk) = chunkIndices{iChunk}(1);
+        indices_LastPacket(iChunk) = chunkIndices{iChunk}(end);
+    end
+    timestamps_FirstPacket = dataTable.timestamp(indices_FirstPacket);
+    timestamps_LastPacket = dataTable.timestamp(indices_LastPacket);
+    timestamp_gaps = timestamps_FirstPacket(2:end) - timestamps_LastPacket(1:end-1); % in seconds
+    
+    % If the gap in timestamp between packets is < 6 seconds, flag this packet;
+    % calculate elapsed time in systemTick
+    
+    % iTimegap + 1 is the index of the chunk which does not need a new
+    % correctedAlignTime calculated (aka chunksWithTimingFromPrevious)
+    elapsed_systemTick = NaN(1,length(timestamp_gaps));
+    
+    for iTimegap = 1:length(timestamp_gaps)
+        % Check if timegap is < 6 seconds and if this chunk was not created
+        % because of change in sampling rate
+        if timestamp_gaps(iTimegap) < 6 && ~ismember(indices_LastPacket(iTimegap),indices_changeFs)
+            chunksWithTimingFromPrevious = [chunksWithTimingFromPrevious iTimegap + 1];
+            systemTick_FirstPacket = dataTable.systemTick(indices_FirstPacket(iTimegap + 1));
+            systemTick_Preceeding = dataTable.systemTick(indices_LastPacket(iTimegap));
+            
+            % Need to use calculateDeltaSystemTick in order to handle situations when
+            % systemTick rollover occurred
+            elapsed_systemTick(iTimegap)= calculateDeltaSystemTick(systemTick_Preceeding,systemTick_FirstPacket);
+        end
+    end 
+end
+
 %%
 % Loop through each chunk to determine offset to apply (as determined by
-% average difference between packetGenTime and expectedElapsed)
+% average difference between packetGenTime and expectedElapsed) --
+% calculated for all chunks here, will subsequently only apply error for
+% chunks which are preceeded by gaps >= 6 seconds
 disp('Determining start time of each chunk')
 
 % PacketGenTime in ms; convert difference to 1e-4 seconds, units of
 % systemTick and expectedElapsed
 diff_PacketGenTime = [1; diff(dataTable.PacketGenTime) * 1e1];
```

Comment:
it is unclear why multiplying by 1e1 (10) - may be add a comment to the right?

Reply:
Resolved with 83db53bba4ad68982cbfeac256da13d0034a9baa